### ISSUE TITLE
examples/resource_partitioner build issue on master branch / ubuntu 18

### ISSUE THREAD

**Issue by prlw1** [STEllAR-GROUP/hpx#3723]
First attempt at building git master branch on ubuntu 18.04.02 (so gcc 7.3.0):
```
cmake -DCMAKE_INSTALL_PREFIX=/home/prlw1/hpx -DHPX_WITH_MALLOC=jemalloc ..
```
(as installed tcmalloc not found)
```
[ 83%] Building CXX object examples/resource_partitioner/CMakeFiles/async_customization.dir/async_customization.cpp.o
In file included from /home/prlw1/src/hpx/examples/resource_partitioner/async_customization.cpp:9:0:
...
/home/prlw1/src/hpx/hpx/runtime/threads/executors/guided_pool_executor.hpp:104:40: error: no match for call to ‘(hpx::threads::executors::pool_numa_hint<dummy_tag>) (hpx::util::tuple<short unsigned int, double>&)’
             int domain = numa_function_(ts...);
                          ~~~~~~~~~~~~~~^~~~~~~
```
(I see that this is in "examples" so will try not to compile them)

**Comment by prlw1** [comment]
-DHPX_WITH_EXAMPLES=OFF got a successful build.

Amusingly on another system (gcc 7.4.0), I see
```
-- C++ mode enforced: C++17
...
CMake Error at cmake/HPX_Message.cmake:47 (message):
  HPX needs support for C++11 std::forward_list
```
yet
```
$ g++ -std=c++17 -Wall -o foo foo.cpp
$ cat foo.cpp 
#include <forward_list>
int main()
{
        std::forward_list<int>foo{1,2,3};
        return 0;
}
```
investigating...

**Comment by biddisco** [comment]
This is a known breakage caused by a recent merge. The error in `guided_pool_executor` will be fixed soon. I have to finish something else fist, then I will resolve this. In the meantime, doing a build using `make -k` will ignore errors and you can at least build the remainder of hpx. Once I've fixed the problem, I'll close this issue.

**Comment by msimberg** [comment]
Fixed by #3736.
